modules/vmdatasource/src/xmpstatsource.cpp

```diff
@@ -0,0 +1,186 @@
+/* 
+ * Copyright 2015 Intel(r) Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http ://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include <algorithm>
+
+#include "xmpstatsource.hpp"
+
+#define VMF_STATISTICS               "statistics"
+
+#define VMF_STAT                     "stat"
+
+#define VMF_STAT_NAME                "name"
+#define VMF_STAT_UPDATE_MODE         "update_mode"
```

Comment:
`update_mode` shouldn't be saved - each `Stat` object should be disabled after loading until user sets new update mode.